backend/models/task.model.ts

```diff
@@ -0,0 +1,48 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+import { TaskStatus } from "../types";
+
+export interface Task extends Document {
+  id: string;
+  title: string;
+  description: string;
+  requiresApproval: boolean;
+  status: TaskStatus;
+  document: string;
+  dueDate: Date;
+  expiry: Date;
+}
+
+const TaskSchema: Schema = new Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    required: true,
+  },
+  requiresApproval: {
+    type: Boolean,
+    required: true,
+  },
+  status: {
+    type: String,
+    required: true,
+    enum: ["Pending", "Completed", "Approved"],
+  },
+  document: {
+    type: String,
+    required: true,
+  },
```

Comment:
Not 100% sure but - should `document` be required? What if a task doesn't involve a document?

Reply:
bro is spittin

Reply:
LOL and now we wait for owen to spit some facts on this one 🧠

Reply:
`document` should be optional as sometimes the task will just include a description of the task or a link. For example, a task might be something like "Update your profile photo".

Reply:
NVM the client wants them to be mandatory